service/producer/kafka.go

```diff
@@ -0,0 +1,19 @@
+package producer
+
+import (
+	"github.com/google/uuid"
+	"github.com/razorpay/metro/internal/config"
+)
+
+type KafkaProducer struct {
+	// will hold the kafka client impl
+}
+
+func NewKakfaProducer(config *config.ConnectionParams) IProducer {
```

Comment:
should there be different implementations of producer for kafka and pulsar? At what level should the segregation of message brokers done?